x/circuit/client/cli/tx.go

```diff
@@ -0,0 +1,146 @@
+		Use:   "authorize [grantee] [permission_level] [limit_type_urls] --from [granter]",
+		Short: "Authorize an account to trip the circuit breaker.",
+		Long: `Authorize an account to trip the circuit breaker.
+		"SOME_MSGS" =     1,
+		"ALL_MSGS" =      2,
+		"SUPER_ADMIN" =   3,`,
+		Example: fmt.Sprintf(`%s circuit authorize [address] 0 "cosmos.bank.v1beta1.MsgSend,cosmos.bank.v1beta1.MsgMultiSend"`, version.AppName),
+		Args:    cobra.RangeArgs(3, 4),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			grantee, err := sdk.AccAddressFromBech32(args[0])
+			if err != nil {
+				return err
+			}
+
+			lvl, err := math.ParseUint(args[1])
+			if err != nil {
+				return err
+			}
+
+			var typeUrls []string
+			if len(args) == 4 {
+				typeUrls = strings.Split(args[2], ",")
+			}
+
+			permission := types.Permissions{Level: types.Permissions_Level(lvl.Uint64()), LimitTypeUrls: typeUrls}
+
+			msg := types.NewMsgAuthorizeCircuitBreaker(clientCtx.GetFromAddress().String(), grantee.String(), &permission)
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+// TripCircuitBreakerCmd returns a CLI command handler for creating a MsgTripCircuitBreaker transaction.
+func TripCircuitBreakerCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "disable [type_url]",
+		Short:   "disable a message from being executed",
+		Example: fmt.Sprintf(`%s circuit authorize trip "cosmos.bank.v1beta1.MsgSend,cosmos.bank.v1beta1.MsgMultiSend"`, version.AppName),
+		Args:    cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := cmd.Flags().Set(flags.FlagFrom, args[0]); err != nil {
+				return err
+			}
+
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			var msgTypeUrls []string
+			if len(args) == 4 {
+				msgTypeUrls = strings.Split(args[0], ",")
+			}
+
+			msg := types.NewMsgTripCircuitBreaker(clientCtx.GetFromAddress().String(), msgTypeUrls)
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+// ResetCircuitBreakerCmd returns a CLI command handler for creating a MsgRestCircuitBreaker transaction.
+func ResetCircuitBreakerCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "reset [type_url]",
+		Short:   "Enable a message to be executed",
+		Example: fmt.Sprintf(`%s circuit authorize reset "cosmos.bank.v1beta1.MsgSend,cosmos.bank.v1beta1.MsgMultiSend"`, version.AppName),
+
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := cmd.Flags().Set(flags.FlagFrom, args[0]); err != nil {
+				return err
+			}
+
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			var msgTypeUrls []string
+			if len(args) == 4 {
```

Comment:
I don't understand this, args can only be 1 because `ExactArgs` is set to one.

Reply:
i removed, the code didnt read like it was needed. @samricotta any reason for this?

Reply:
I think this one might have been yours @tac0turtle but I don't see reason for this